go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -264,17 +264,19 @@ func (c *Controller) syncServices(key string) error {
 				family = v1.IPv6Protocol
 			}
 			eps := getLbEndpoints(endpointSlices, svcPort, family)
-			// Reconcile OVN
+			// Reconcile only if it has endpoints, otherwise it will match the default reject rule for services.
+			// It also saves OVN churn adding and deleting a load balancer without endpoints.
+			if len(eps) == 0 {
+				klog.V(4).Infof("Service %s/%s without endpoints", name, namespace)
+				continue
```

Comment:
don't we need to do work here if the service goes from 1 endpoint to 0?

Reply:
yeah, this was a premature optimization, I have to remove the commit and  always update the loadbalancer